default_http_headers not parsed as strings

**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
2025.8.3+2e62eb5 -- vanilla, possible bug was introduced prior

**How did you install SearXNG?**
docker compose container

**What happened?**
Docker container hangs, granian exits with type mismatch error, no traffic served

**How To Reproduce**
In settings.yml:
```
server:
  default_http_headers:
    X-XSS-Protection: 0
```

**Expected behavior**
Granian serves requests, adding default_http_headers to responses

**Screenshots & Logs**
```
searxng  | Process granian-worker:
searxng  | Traceback (most recent call last):
... cut for brevity ...
searxng  |   File "/usr/local/searxng/searx/webapp.py", line 1390, in static_headers
searxng  |     headers[header] = value
searxng  |     ~~~~~~~^^^^^^^^
searxng  |   File "/usr/lib/python3.12/wsgiref/headers.py", line 56, in __setitem__
searxng  |     (self._convert_string_type(name), self._convert_string_type(val)))
searxng  |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
searxng  |   File "/usr/lib/python3.12/wsgiref/headers.py", line 45, in _convert_string_type
searxng  |     raise AssertionError("Header names/values must be"
searxng  | AssertionError: Header names/values must be of type str (got 0)
searxng  | [ERROR] Unexpected exit from worker-1
searxng  | [INFO] Shutting down granian
```

**Additional context**
default_http_headers key/value pairs should always be strings
this behavior was not present prior to the switchover to granian
Originally posted a band-aid fix in the SearXNG public instance owners Matrix room, asked to file a bug report! Thank you!

--- a/searx/webapp.py
+++ b/searx/webapp.py
@@ -1387,7 +1387,8 @@ def static_headers(headers: Headers, _path: str, _url: str) -> None:
     headers['Cache-Control'] = 'public, max-age=30, stale-while-revalidate=60'
 
     for header, value in settings['server']['default_http_headers'].items():
-        headers[header] = value
+        # cast value to string, as WhiteNoise requires header values to be strings
+        headers[header] = str(value)
 
 
 app.wsgi_app = WhiteNoise(

